src/mlx5_regex.c

```diff
@@ -74,6 +177,155 @@ static void print_raw(void *ptr, size_t size)
 	}
 }
 
+static inline int
+regex_get_pdn(void *pd, uint32_t *pdn)
+{
+	struct mlx5dv_obj obj;
+	struct mlx5dv_pd pd_info;
+	int ret = 0;
+	obj.pd.in = pd;
+	obj.pd.out = &pd_info;
+	ret = mlx5dv_init_obj(&obj, MLX5DV_OBJ_PD);
+	if (ret) {
+                syslog(LOG_ERR, "Failed to get PD object info\n");
+		//DRV_LOG(DEBUG, "Fail to get PD object info");
+		return ret;
+	}
+	*pdn = pd_info.pdn;
+	return 0;
+}
+
+struct ibv_pd *
+regex_alloc_pd(void *ctx)
+{
+	//struct mlx5dv_obj *ppd;
+
+        //ppd = calloc(1, sizeof(*ppd));
+        return ibv_alloc_pd(ctx);
+}
+
+static int
+rxp_create_mkey(struct mlx5_regex_ctx *ctx, void *ptr, size_t size,
+	uint32_t access, struct mlx5_regex_mkey *mkey)
+{
+	uint32_t pdn;
+	struct mlx5_devx_mkey_attr mkey_attr;
+	int ret;
+        struct mlx5_devx_obj *tmp_mkey;
+        void *mkc;
+        size_t pgsize ;
+        uint32_t translation_size;
+	uint32_t in[DEVX_ST_SZ_DW(create_mkey_in)] = {0};
+        uint32_t out[DEVX_ST_SZ_DW(create_mkey_out)] = {0};
+
+        pgsize = sysconf(_SC_PAGESIZE);
+        if (pgsize == (size_t) -1) {
+                syslog(LOG_ERR, "Failed to get page siza\ne");
+		return -ENOMEM;
+        }
+        tmp_mkey = calloc(1, sizeof(*tmp_mkey));
+
+        syslog(LOG_ERR, "GERRY: tmp_mkey = %p\n", tmp_mkey);
+
+	if (!tmp_mkey) {
+                syslog(LOG_ERR, "Failed to allocate memory!\n");
+		return -ENOMEM;
+	}
+
+	/* Register the memory. */
+        mkey->umem = mlx5dv_devx_umem_reg(ctx->ibv_ctx, ptr, size,access);
+
+	if (!mkey->umem) {
+                syslog(LOG_ERR, "Failed to register memory!\n");
+		return -ENODEV;
+	}
+	/* Create mkey */
+	ret = regex_get_pdn(ctx->pd, &pdn);
+	if (ret < 0) {
+                syslog(LOG_ERR, "Failed to get pdn!\n");
+		return -ENODEV;
+	}
+
+        syslog(LOG_ERR, "GERRY: mkey->umem->umem_id = 0x%x\n", (mkey->umem)->umem_id);
+
+	mkey_attr = (struct mlx5_devx_mkey_attr) {
+		.addr = (uintptr_t)ptr,
+		.size = (uint32_t)size,
+		.umem_id = (mkey->umem)->umem_id,
+		.pg_access = 1,
+		.umr_en = 0,
+		.pd = pdn,
+	};
+
+translation_size = (RTE_ALIGN(mkey_attr.size, pgsize) * 8) / 16;
```

Comment:
Please keep a proper coding style.
See DOCA coding style, and script
https://wikinox.mellanox.com/display/SW/DOCA+coding+style